internal/controller/argocd/api/v1alpha1/application_types.go

```diff
@@ -29,6 +30,11 @@ type ApplicationSource struct {
 	Helm           *ApplicationSourceHelm      `json:"helm,omitempty"`
 	Kustomize      *ApplicationSourceKustomize `json:"kustomize,omitempty"`
 	Chart          string                      `json:"chart,omitempty"`
+	Path           string                      `json:"path,omitempty"`
+}
+
+func (a Application) IsMultisource() bool {
+	return a.Spec.Sources != nil && len(a.Spec.Sources) > 0
```

Comment:
The nil check is not necessary since len of a var with a nil value is 0.

With the complexity of checking whether an App is multisource reduced to nothing more than a _very straightforward_ check that the len of sources is not 0, I would perhaps not even bother with this helper function, but it's not a hill I would die on.

Reply:
I've no strong preference at this point, removing.